services/plugin/src/Plugin.tsx

```diff
@@ -118,68 +113,81 @@ export const Plugin = ({
-                        setCommunicationReceived(true)
-                        return iframeProps
-                    }
-                )
-                return () => listener.cancel()
-            }
-
-            // if iframe has sent initial request, send new props
-            if (
-                communicationReceived &&
-                iframeRef.current.contentWindow &&
-                !inErrorState
-            ) {
-                postRobot
-                    .send(
-                        iframeRef.current.contentWindow,
-                        'updated',
-                        iframeProps
-                    )
-                    .catch((err) => {
-                        // log postRobot errors, but do not bubble them up
-                        console.error(err)
-                    })
-            }
+        const iframeProps = {
+            ...memoizedPropsToPass,
+            alertsAdd,
+            // If a dimension is either specified or container-driven,
+            // don't send a resize callback to the plugin. The plugin can
+            // use the presence or absence of these callbacks to determine
+            // how to handle sizing inside
+            setPluginHeight: !height ? setPluginHeight : null,
+            setPluginWidth: !width && clientWidth ? setPluginWidth : null,
+            setInErrorState,
+            clientWidth,
+        }
+
+        // If iframe has not sent initial request, set up a listener
+        // (this will still be set up if plugin needs to reload)
+        if (!communicationReceivedRef.current && !inErrorState) {
+            const listener = postRobot.on(
+                'getPropsFromParent',
+                // listen for messages coming only from the iframe rendered by this component
+                { window: iframeRef.current.contentWindow },
+                (): any => {
+                    communicationReceivedRef.current = true
+                    return iframeProps
+                }
+            )
+            return () => listener.cancel()
+        }
+
+        // If iframe has sent initial request, send new props
+        if (
+            communicationReceivedRef.current &&
+            iframeRef.current.contentWindow &&
+            !inErrorState
+        ) {
+            postRobot
+                .send(iframeRef.current.contentWindow, 'updated', iframeProps)
+                .catch((err) => {
+                    // log postRobot errors, but do not bubble them up
+                    console.error(err)
+                })
         }
-        // prevCommunicationReceived update should not retrigger this hook
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [memoizedPropsToPass, communicationReceived, inErrorState, alertsAdd])
+    }, [
+        memoizedPropsToPass,
+        inErrorState,
+        // The following should be pretty stable:
+        alertsAdd,
+        clientWidth,
+        height,
+        width,
```

Comment:
Added these three after removing the `eslint-disable` (refactoring to use refs made state management simpler)